source/jsCalendar.js

```diff
@@ -97,6 +97,7 @@ var jsCalendar = (function(){
         this._now = null;
```

Comment:
Move the code to separate extension file, maybe named `jsCalendar.eventmarks.js`

Based on your code, extension code :

```javascript
(function($){
	// Check if jsCalendar exists
	if (!$) throw 'Error jsCalendar library was not found';

	// Example Extension
	var init = function(instance) {
		instance._eventMarks = {};
	};
	var update = function(instance, month) {
		for (var i = month.days.length - 1; i >= 0; i--) {
			// If date is marked for events
			var cellContent = instance._elements.bodyCols[i].textContent;
			if (month.days[i].getTime() in instance._eventMarks) {
				cellContent += '<div class="jsCalendar-event-marks">'
				var moreColors = 0;
				var colors = instance._eventMarks[month.days[i].getTime()];
				for (var j = 0; j < colors.length; j++) {
					if (j < 3) {
						cellContent += '<div class="jsCalendar-event-mark" style="background-color: ' + colors[j] + '"></div>';
					}
					else moreColors++;
				}
				if (moreColors > 0) {
					cellContent += '<div class="jsCalendar-event-more-mark">+' + moreColors + '</div>';
				}
				cellContent += '</div>';
				instance._elements.bodyCols[i].innerHTML = cellContent;
			}
		}
	};

	$.prototype._addEventMarksForDate = function(date, eventColors) {
		date = $.tools.parseDate(date);
		date.setHours(0, 0, 0, 0);
		date = date.getTime();

		if (!(date in this._eventMarks)) {
			this._eventMarks[date] = [];
		}

		this._eventMarks[date] = this._eventMarks[date].concat(eventColors);
	};

	$.prototype._removeEventMarksForDate = function(date, eventColors) {
		date = $.tools.parseDate(date);
		date.setHours(0, 0, 0, 0);
		date = date.getTime();

		if (date in this._eventMarks) {
			for (var i = 0; i < eventColors.length; i++) {
				var pos = this._eventMarks[date].indexOf(eventColors[i]);
				if (pos >= 0) {
					this._eventMarks[date].splice(pos, 1);
				}
			}
		}
	};

	$.prototype._clearEventMarksForAllDates = function() {
		this._eventMarks = {};
	};

	// Select dates
	$.prototype.addEventMark = function(date, eventColors){
		// If no arguments
		if (typeof date === 'undefined') {
			// Return
			return this;
		}

		// If eventColors not array
		if (!(eventColors instanceof Array)) {
			// Lets make it an array
			eventColors = [eventColors];
		}
		// Mark dates
		this._addEventMarksForDate(date, eventColors);
		// Refresh
		this.refresh();

		// Return
		return this;
	};

	// Select dates
	$.prototype.removeEventMark = function(date, eventColors){
		// If no arguments
		if (typeof date === 'undefined') {
			// Return
			return this;
		}

		// If eventColors not array
		if (!(eventColors instanceof Array)) {
			// Lets make it an array
			eventColors = [eventColors];
		}
		// Unmark dates
		this._removeEventMarksForDate(date, eventColors);
		// Refresh
		this.refresh();

		// Return
		return this;
	};

	// Clear event marks from all dates
	$.prototype.clearAllEventMarks = function(){
		// Unselect all dates
		this._clearEventMarksForAllDates();
		// Refresh
		this.refresh();

		// Return
		return this;
	};

	// Load extension
	$.ext('EventMarks', {
		init : init,
		update : update
	});

})(window.jsCalendar);
```